### ISSUE TITLE
Walkthrough Findings - Kickoff

### ISSUE THREAD

**Issue by chaseakirby** [department-of-veterans-affairs/va.gov-team#41201]
## User Story
_"As a VFS team member and Governance team member, I want clear and up to date documentation and guidance so that I can easily perform my role in the Collaboration Cycle.

- Intended Audience: VFS Team and Governance Team
- Desired Outcome: Updated documentation and templates
- Due Date: 

## Description
Tasks listed below are from our [Core CC Walkthrough session notes](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2183757962/Notes+-+Core+Collaboration+Cycle+ticket+Walk-through)
Remaining open Opportunity statements are also noted. 
[Research Findings spreadsheet (Pre Launch Improvements tab)](https://docs.google.com/spreadsheets/d/1_EGjrHtr3gEam72K4zEoFMdkQLBIvIjQ_lmes-RgWLw/edit#gid=1309674612)

Additional background for [Collaboration Cycle - Improve tracking and status visibility of VA.gov products](https://app.zenhub.com/workspaces/platform-governance-team-5f85b91c14d8df0018fac414/issues/department-of-veterans-affairs/va.gov-team/34697) initiative <br />

## Artifacts Needing Updates
[Kickoff Moderator Responsibilities](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2169274464) 
[Kickoff Ticket Template](https://github.com/briandeconinck/cc-test/issues/new?assignees=briandeconinck&labels=collab-cycle-review%2Ccollaboration-cycle%2Cgovernance-team&template=core-collab-cycle-yaml.yml&title=Collaboration+Cycle+for+%5BTeam+Name%2C+Product+Name%2C+Feature+Name%5D) 
[Platform Website: Kickoff Touchpoint Page](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2192998401/Cheryl+Brian+-+Copy+of+Collaboration+Cycle+Kickoff)
[Collaboration Cycle Tracker](https://docs.google.com/spreadsheets/d/1OgPyEvUlNF6EnaYMFAXJkV6FKOvZnlPnbOQ2fAJ7W7A/edit#gid=1116695886)
[Previous Collab Cycle Kickoff GitHub Template](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/.github/ISSUE_TEMPLATE/collab-cycle-kickoff.md)

## Tasks
  - [x] (Final Walkthrough) FW: ZenHub board: Change filter to CC-Request

- [x] [Platform Website: Kickoff Touchpoint Page](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2192998401/Cheryl+Brian+-+Copy+of+Collaboration+Cycle+Kickoff)
  - [x] Create draft page
  - [x] Update intro text to remove commenting in GitHub
  - [x] Check whether questions 1 - 13 should follow Kickoff questions in yml
  - [x] Remove VA Business Owner question
  - [x] Review content on [CC page](https://vfs.atlassian.net/wiki/spaces/CC/overview?homepageId=1524891996)
  - [x] Opportunity Statement: Match list of required questions to YAML ticket
    - [x] Opportunity Statement: Evaluate required questions
  - [x] Add Artifacts - Product Outline ~~& Other Artifacts~~ to match GitHub template
  - [x] Opportunity Statement: Update to clearly detail needed information prior to Kickoff (updating required questions may take care of this)
  - [x] Opportunity Statement: Add guidance detailing that it is the PMs responsibility to attach artifacts (Product Outline)
  - [ ] Opportunity Statement: Add directions for attaching artifacts and a description of the Product Outline (Not now)
     - Per Gov team meeting on 6/14, this will not be addressed as there is no need 
  - [x] [Product Outline template](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/product-management/product-outline-template.md) linked

  - [x] FW: Update link in "asynchronous Collaboration Cycle Kickoff" to new ticket
  - [x] FW: Publish new GitHub template first
  - [x] FW: "We'll update the Collaboration Cycle Request ticket with recommended touchpoints and respond in the #vfs-slack-support request post"
  - [x] FW: Remove "determine which practice areas" - Tech-debt work
  - [x] FW: Question 3: Add "Other" to match template
  - [x] FW: Product Outline link needs changed to link being used in template

- [x] [Kickoff Ticket Template](https://github.com/briandeconinck/cc-test/issues/new?assignees=briandeconinck&labels=collab-cycle-review%2Ccollaboration-cycle%2Cgovernance-team&template=core-collab-cycle-yaml.yml&title=Collaboration+Cycle+for+%5BTeam+Name%2C+Product+Name%2C+Feature+Name%5D)
  - [x] Remove "yml" and "core" - maybe rename to "Collaboration Cycle Requests"
  - [x] Intro paragraph changes
    - [x] Paragraph 1 - Remove each practice area
    - [x] Paragraph 2 - Remove "which practice areas"
    - [x] Asterisk text - Move to new line near first field and try to make red
  - [x] VFS product information - Make header H2
    - [x] Consider removing note
    - [x] Change "start date of enhancement" to "when did/will you start working on this feature?"
  - [x] Kickoff Questions changes
    - [x] Rename to match [CC Kickoff](https://vfs.atlassian.net/wiki/spaces/CC/pages/1760493650/Collaboration+Cycle+Kickoff) "feature" vs "product"
    - [x] Set header as H2
    - [x] Update link of "Getting started with the Collaboration Cycle" to "Collaboration Cycle Kickoff"
    - [x] Remove "other" option from "Will your work result in visible changes to the user experience?"
    - [x] Set 3rd option as "Unsure" for "Are you doing research with VA.gov users?"
    - [x] Set 3rd option as "other" for "Will your work involve changes to..."
      - [x] Add "**For information and examples of Static vs Dynamic pages, review the Static vs Dynamic guidance"
        - [x] Link text to https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/teams/sitewide-content/static-and-non-static-page-documentation.md
        - [x] follow up on the “other” or referencing alerts or emails. but for now default to how current, if not current would need to update the flow - Need to check with Clayton context for this
  - [x] Google Analytics/Domo question
    - [x] Set 3rd option as "unsure" for "Does your product or feature have Google Analytics tracking and a KPI dashboard in Domo?"
    - [x] Remove "or" and add "/" from "Does your product or feature have Google Analytics..."
    - [x] Check if both Google Analytics and KPI are both required if one is there
  - [x] Set last option as "unsure" for "If yes, do you need to capture any other analytics or metrics in addition to what you're currently tracking?"
    - [x] Investigate if we can remove "None" from this dropdown
  - [x] Remove "Who is the VA business owner for this product?" question and dropdown
  - [x] Set "Artifacts" header as H2
    - [ ] Add another option for attaching artifacts
       - Per Gov team meeting on 6/14, the team decided this option is unnecessary at this time.
  - [x] Add new question section for "instructions" - Should include text at top of current kickoff ticket to notify us in slack
  - [x] Research GitHub/Slack integration
  - [x] Investigate putting kickoff questions into an accordion
  - [x] Opportunity Statement: Evaluate required questions
    - [x] Update YAML if required questions change
  - [x] Opportunity Statement: Link to Platform Website guidance on Kickoff artifacts (Product Outline template link may be sufficient, may be worth discussing)

  - [x] FW: Change title to "Collaboration Cycle Request"
  - [x] FW: Add labels: collab-cycle-review, governance-team, collaboration-cycle, CC-Request
  - [x] FW: First sentence "VA.gov experience standard" to "VA.gov Experience standards"
  - [x] FW: Feature name should be required
  - [x] FW: Change Next Steps "Notify ... in #vfs-platform support slack channel"
  - [x] FW: Copy formatting from other Requests for "Sitewide Content & IA Intake Request"
  - [x] FW: Remove links to other touchpoint tickets (should only refer to PW page for respective touchpoints)
  - [x] FW: Update Privacy and Security Review to "Privacy, Security, and Infrastructure Readiness Review"
  - [x] FW: Privacy and Security Review: Platform Action: Checkbox "Privacy, Security, and Infrastructure Readiness Review has been completed"

- [ ] [Kickoff Moderator Responsibilities](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2169274464)
  - [x] Add instructions to convert ticket to an Epic
  - [x] Step 1 - Update to notify team in slack, not as a comment on ticket
  - [x] Step 2 
    - [x] Fix Custom Tracks Flow link
    - [x] Look into removing "existing feature question" from flow (Chase and I discussed.  Think this would work better as 'TechDebt'.  Won't affect release of ticket.)
      - Not currently asked, should be able to tell if it is live on VA.gov
  - [x] Step 3 
    - [x] [Collaboration Point Tracker:] 
(https://docs.google.com/spreadsheets/d/1OgPyEvUlNF6EnaYMFAXJkV6FKOvZnlPnbOQ2fAJ7W7A/edit#gid=1116695886) Add "Feature Label" column
    - [x] Collaboration Point Tracker: Add "CC Request Ticket Column"
    - [x] Collaboration Point Tracker: Add divider between new columns and Product Manager - No longer need to track anything to the right
    - [x] Remove Product Manager, Collab Cycle Kickoff Date, Collab Cycle GitHub Link, and "Insert an X on all of the Collaboration Cycle touchpoints that are not recommended" (Remove from Confluence operational documentation only.)
  - [x] Step 4 - Update to respond in slack
  - [x] Step 5 - Remove instructions to "Check off the below checkboxes..."

  - [x] FW: Step 1: Remove link in "tag PM in the slack platform-collaboration-cycle channel"
  - [x] FW: Step 2: Change "tag PM in #vfs-platform-support request post. If they did not post, tag PM in their team slack channel."
  - [x] FW: Step 3: Remove Product Manager
  - [x] FW: Step 3: Change to "Collaboration Cycle Request date" and "Collaboration Cycle Request GitHub link"
  - [x] FW: Step 3: Remove step "Insert X..."
  - [x] FW: Step 4: "Once we determined the necessary touchpoints, edit the Collaboration Cycle Request Ticket to include recommended touchpoints"
  - [x] FW: Step 4: Add code blocks of each touchpoint for easy copy and pasting "If Design Intent was recommended, insert this code into the body of the Collaboration Cycle Request"
  - [x] FW: Step 4: "When we add all the recommended touchpoints, Reply in slack request post "Governance team has updated the Collaboration Cycle Request with recommended touchpoints" 
    - [x]  If they did not post, tag PM in their team slack channel.
  - [x] FW: Step 4: Remove "For those who started the CC process during/after..."
  - [x] FW: Step 4: Remove "For those who started the CC process prior..."
  - [x] FW: Step 5: change reply to "We have added recommended touchpoints, please let us know if you have any questions" (paraphrased)
  - [x] FW: Step 8: Add "Post to #collaboration-cycle channel" (like currently do for other touchpoints)
  - [ ] FW: Will need to notify Security team to update Collaboration Cycle Request ticket

- [x] Once new ticket is ready for release, delete [Previous Collab Cycle Kickoff GitHub Template](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/.github/ISSUE_TEMPLATE/collab-cycle-kickoff.md)

## Acceptance Criteria
- [x] [Kickoff Moderator Responsibilities](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2169274464) has been updated
- [x] [Kickoff Ticket Template](https://github.com/briandeconinck/cc-test/issues/new?assignees=briandeconinck&labels=collab-cycle-review%2Ccollaboration-cycle%2Cgovernance-team&template=core-collab-cycle-yaml.yml&title=Collaboration+Cycle+for+%5BTeam+Name%2C+Product+Name%2C+Feature+Name%5D) has been updated
- [x] [Platform Website: Kickoff Touchpoint Page](https://vfs.atlassian.net/wiki/spaces/CC/pages/1760493650/Collaboration+Cycle+Kickoff) has been updated
- [x] [Previous Collab Cycle Kickoff GitHub Template](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/.github/ISSUE_TEMPLATE/collab-cycle-kickoff.md) has been deleted when ready to release new ticket
- [x] Collaboration Cycle Tracker has been updated
- [x] Updates have been Peer Reviewed by Governance Team member
- [x] User Story need has been fulfilled

---

## Related Governance Team Resources
_For additional information and guidance to complete tasks, please see links below._
[Documentation Update Guidance for GitHub templates](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2047868956/Guidance+for+updating+GitHub+Issue+templates) <br />
[Documentation Update Guidance for Platform Website pages](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2089091115/Guidance+for+updating+Platform+Website+pages) <br />
[Documentation Update Guidance for Confluence pages](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2089222173/Guidance+for+updating+Confluence+pages) <br />
[Guidelines for Governance Team Confluence Space](https://vfs.atlassian.net/wiki/spaces/AP/pages/1127219219/Guidelines+for+Team+spaces) <br />
[Guidelines for Platform Products and Services Confluence Space](https://vfs.atlassian.net/wiki/spaces/AP/pages/1321140361/Guidelines+for+Platform+Products+and+Services+space) <br />
[Description of Platform Website spaces and content](https://vfs.atlassian.net/wiki/spaces/AP/pages/1241579796/Organization) <br />

---
## How to configure this issue
- [ ] **Ensure that story is attached to appropriate Iniative (if not tech debt)** (what body of work is this a part of?)
- [ ] Ensure that any additional labels are added ('tech-debt', 'collaboration-cycle')
- [ ] Ensure that story is in the correct pipeline (Backlog, Planned, In Progress, Review/QA, Closed)
- [ ] Ensure that assignees are correct so you receive all updates/comments on the ticket

**Comment by briandeconinck** [comment]
@shiragoodman @chaseakirby, some notes on the GitHub/Slack integration. Short version: this is going to be really easy to set up.

The DSVA Slack already has the official GitHub integration set up, so no need to install/get permission for anything extra.

Whoever sets up the notifications will have to connect their GitHub account to Slack. This is a super easy process, I did it with my account in ten seconds. I'm fine being the one to set up the notifications if that makes things easier. However, I'm not sure what happens to the notifications if that person leaves and their account disappears but my guess is the notifications will stop. For whoever sets it up, that's something to try to remember when they eventually offboard.

Once the account is connected:

(1) In the channel where you want notifications, subscribe to the repo with the slash-command: `/github subscribe department-of-veterans-affairs/va.gov-team`. This subscribes the channel to everything happening in the repo.

(2) Next, remove the stuff we don't want:
- `/github unsubscribe department-of-veterans-affairs/va.gov-team pulls`
- `/github unsubscribe department-of-veterans-affairs/va.gov-team commits`
- `/github unsubscribe department-of-veterans-affairs/va.gov-team releases`
- `/github unsubscribe department-of-veterans-affairs/va.gov-team deployments`

(3) At this point, the channel is subscribed to all issues. We can filter that down with: `/github subscribe department-of-veterans-affairs/va.gov-team +label:"name-of-label"`

So long as the label we're filtering on is referenced in the Yaml template, every time the Yaml form generates a new issue the GH integration will post a notification in the channel. We can have more than one label in the Yaml template, so for notifications we can use an existing label or create a new one just for this purpose.

Notifications are verbose, a lot of text gets passed to Slack. I think there is some character limit but I don't think there's anything we can do to set it ourselves.

Here's a screenshot of a notification from an issue in the briandeconinck/cc-test repo.

![slack-notification-new-issue.png](https://images.zenhubusercontent.com/61a66f92c1713a0111e338b8/76c863c5-c3f3-4ea1-93e6-e75302172d10)

Notes from some quick testing:
- Adding the label to an existing issue doesn't seem to generate a notification. Notifications only seem to show up when a new issue has the label at the moment it's created. (Shouldn't be an issue for the Yaml form, but something to be aware of in case workflows change.)
- Comments and most changes to the issues DO NOT generate notifications. That's probably desired behavior.
- Closing an issue and re-opening a closed issue DO generate a notification. That's probably not a bad thing, I think? Closing the issue would presumably mark the end of a trip through the CC, and if it gets re-opened by the PM it's probably good to know about that too. Either way, that notification is short and sweet.

![slack-notification-close-reopen.png](https://images.zenhubusercontent.com/61a66f92c1713a0111e338b8/2014f335-3773-4bd2-9835-fb14d3ae3159)

So for next steps on notifications, I think we just need to decide:
- Whose account sets up the notifications?
- What channel should display the notifications? (I'm guessing #platform-collaboration-cycle)
- What label do we want to subscribe to?

Let me know if there's anything specific you'd like me to test, or if you'd like to see a live demo.

**Comment by shiragoodman** [comment]
re: the gh/Slack integration - awesome, awesome, awesome. I'm 100% for this, however I don't think we need to rush to get it in prior to the launch. thank you, Brian!

@chaseakirby can you create a ticket for the backlog for implementing Slack notifications for new Collaboration Cycle tickets?

**Comment by briandeconinck** [comment]
For this item:

>  Add another option for attaching artifacts

In a team discussion on 5/18, we determined we would **not** add another field for additional artifacts. We don't want/need prototypes at kickoff and this would mostly just be extra clutter. Leaving as is and marking as complete.

**Comment by briandeconinck** [comment]
Another note on another item:

> Investigate putting kickoff questions into an accordion

I've tried a few different ways of doing this, but no luck. The HTML renders in a "Markdown"-type field object, but GitHub auto-closes any unclosed tags --- so you can't open a `<details>` element in one object, add a bunch of input objects, and then close a `</details>` in another object.

My proposed alternative would be to add a step to the kickoff moderator script to insert the `<summary>` and `<details>` tags to accordion-ify the kickoff questions prior to Design Intent. It's relatively straightforward, just an extra thing for the moderator to have to do.

Thoughts @CherylEvans @chaseakirby @shiragoodman?

**Comment by shiragoodman** [comment]
@briandeconinck im supportive of your proposed alternative.
